contrib/win32/win32compat/w32-sshfileperm.c

```diff
@@ -247,15 +253,96 @@ check_secure_folder_permission(const wchar_t* path_utf16, int read_ok)
 			continue;
 		}
 		else {
-			ret = -1;
+			/* collect all SIDs with write permissions */
+			wchar_t resolved_trustee[UNLEN + 1] = L"UNKNOWN", resolved_trustee_domain[DNLEN + 1] = L"UNKNOWN";
+			DWORD resolved_trustee_len = _countof(resolved_trustee), resolved_trustee_domain_len = _countof(resolved_trustee_domain);
+			SID_NAME_USE resolved_trustee_type;
+
+			ret = -1; // set ret to -1 to indicate that there are bad permissions so message will be logged
+
+			if (log_msg != NULL &&
+				LookupAccountSidW(NULL, current_trustee_sid, resolved_trustee, &resolved_trustee_len,
+				resolved_trustee_domain, &resolved_trustee_domain_len, &resolved_trustee_type) != 0) {
+				if (is_first) {
+					_snwprintf_s(log_msg, log_msg_len, _TRUNCATE, L"%ls\\%ls", resolved_trustee_domain, resolved_trustee);
+					is_first = FALSE;
+				}
+				else {
+					size_t currentLength = wcslen(log_msg);
+					size_t userLength = resolved_trustee_domain_len + 1 + resolved_trustee_len + 2; // +1 for '\\' and +2 for ', '
+					if (wcslen(log_msg) + userLength + 1 > log_msg_len) { // +1 for null terminator
+						log_msg_len *= 2;
+						wchar_t* temp_log_msg = (wchar_t*)malloc(log_msg_len * sizeof(wchar_t));
+						if (temp_log_msg == NULL) {
+							break;
+						}
+						wcscpy_s(temp_log_msg, log_msg_len, log_msg);
+						if (log_msg)
+							free(log_msg);
```

Comment:
Braces make the code more clear and also avoids problems with future changes

```suggestion
						if (log_msg) {
							free(log_msg);
				                }
```

Reply:
I get that but on other hand, this follows the existing convention in the code. Also, this is in the win32 compat project so future merge conflicts due to pulling in upstream changes are not a concern here.

Reply:
I think since it's in the win32 compat only side, it makes a stronger case to fix it throughout the code (separate PR for changes outside the scope of this fix).